contracts/strategies/StrategyProxy.sol

```diff
@@ -83,11 +83,12 @@ contract StrategyProxy {
     function deposit(address _gauge, address _token) external {
         uint256 _balance = IERC20(_token).balanceOf(address(this));
         IERC20(_token).safeTransfer(address(proxy), _balance);
-
         _balance = IERC20(_token).balanceOf(address(proxy));
+
         proxy.execute(_token, 0, abi.encodeWithSignature("approve(address,uint256)", _gauge, 0));
         proxy.execute(_token, 0, abi.encodeWithSignature("approve(address,uint256)", _gauge, _balance));
-        proxy.execute(_gauge, 0, abi.encodeWithSignature("deposit(uint256)", _balance));
+        (bool success, ) = proxy.execute(_gauge, 0, abi.encodeWithSignature("deposit(uint256)", _balance));
```

Comment:
hmm i may be missing something, but shouldn't all proxy calls be checked for success? i see `withdraw()` and `harvest()` have some proxy calls, if some of those low level calls failed in the same way, what is the consequence?

Reply:
Looks good, only need to validate last call for success in out of gas context